File: app/src/main/java/com/example/applligent/nagoriengineering/MyNagoriApplication.java
package com.example.applligent.nagoriengineering;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.arch.persistence.room.Room;
import android.os.Build;

import com.google.firebase.FirebaseApp;
import com.google.firebase.messaging.FirebaseMessaging;

public class MyNagoriApplication extends Application {
    public static final String reminderChannel = "reminder";
    public static final String chatChannel = "Nagori_chat";

    final static String DATABASE_NAME = "db-newNagori";
    private static MyDatabase myDatabase;
    private static MyNagoriApplication myNagoriApplication;

    public static MyDatabase getDatabase() {
        if (myDatabase == null) {
            myDatabase = Room.databaseBuilder(myNagoriApplication,
                    MyDatabase.class, DATABASE_NAME)
                    .allowMainThreadQueries()
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return myDatabase;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        FirebaseApp.initializeApp(this);
        FirebaseMessaging.getInstance().subscribeToTopic("notifications");
        myNagoriApplication = this;
        createNotificationChannels();
    }

    public void createNotificationChannels() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel nagori_chat = new NotificationChannel(
                    chatChannel, "nagori chat",
                    NotificationManager.IMPORTANCE_HIGH);
            nagori_chat.setDescription("new message");

            NotificationChannel reminder = new NotificationChannel(
                    reminderChannel, "reminders",
                    NotificationManager.IMPORTANCE_HIGH);
            reminder.setDescription("new reminder");
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(nagori_chat);
            manager.createNotificationChannel(reminder);
        }
    }
}

File: app/src/main/java/com/example/applligent/nagoriengineering/MyNagoriApplication.kt
package com.example.applligent.nagoriengineering

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.arch.persistence.room.Room
import android.content.Context
import android.os.Build

import com.google.firebase.FirebaseApp
import com.google.firebase.messaging.FirebaseMessaging

class MyNagoriApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        FirebaseApp.initializeApp(this)
        FirebaseMessaging.getInstance().subscribeToTopic("notifications")
        FirebaseMessaging.getInstance().subscribeToTopic("reminders")
        applicationContext
        createNotificationChannels()
    }

    fun createNotificationChannels() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val nagori_chat = NotificationChannel(
                    chatChannel, "nagori chat",
                    NotificationManager.IMPORTANCE_HIGH)
            nagori_chat.description = "new message"

            val reminder = NotificationChannel(
                    reminderChannel, "reminders",
                    NotificationManager.IMPORTANCE_HIGH)
            reminder.description = "new reminder"
            val manager = getSystemService(NotificationManager::class.java)
            manager.createNotificationChannel(nagori_chat)
            manager.createNotificationChannel(reminder)
        }
    }

    companion object {
        val reminderChannel = "reminder"
        val chatChannel = "Nagori_chat"

        internal val DATABASE_NAME = "db-newNagori"
        private var myDatabase: MyDatabase? = null

        fun getDatabase(context: Context): MyDatabase {
            if (myDatabase == null) {
                myDatabase = Room.databaseBuilder(context,
                        MyDatabase::class.java, DATABASE_NAME)
                        .allowMainThreadQueries()
                        .fallbackToDestructiveMigration()
                        .build()
            }
            return myDatabase as MyDatabase
        }
    }
}
